### PR TITLE
Feat: Add expand and vulcan tower

### PR DESCRIPTION
**Pull request by localpthebest**
I'm sorry to whoever has to look at this dogshit

now with 100% more tower and 100% less detek

### CODE REVIEW

**Comment by 1zun4** [comment]
Good luck

**Comment by localpthebest** [comment]
izuna why would you do me like that :cry:

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt [hunk 1]
```diff
@@ -607,4 +632,65 @@ object ModuleScaffold : Module("Scaffold", Category.WORLD) {
...
+    @Suppress("unused")
+    private val onNetworkTick = handler<PlayerNetworkMovementTickEvent> { event ->
+        if (event.state == EventState.PRE) {
+            for(i in 1..expandLenght) {
+                val block = findBestBlockPlacementTarget(
+                    player.blockPos.add(
+                        (-sin(player.yaw.toRadians()) * i).toInt(),
+                        -(player.blockPos.y - placementY),
+                        (cos(player.yaw.toRadians()) * i).toInt()
+                    ),
```
Comment (superblaubeere27) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt]: Pretty smart solution 👍🏻

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt [hunk 2]
```diff
@@ -607,4 +632,65 @@ object ModuleScaffold : Module("Scaffold", Category.WORLD) {
...
+                    BlockPlacementTargetFindingOptions(
+                        listOf(Vec3i(0, 0, 0)),
+                        player.activeItem,
+                        CenterTargetPositionFactory,
+                        BlockPlacementTargetFindingOptions.PRIORITIZE_LEAST_BLOCK_DISTANCE,
+                        player.blockPos.add(
+                            (-sin(player.yaw.toRadians()) * i).toInt(),
+                            -(player.blockPos.y - placementY),
+                            (cos(player.yaw.toRadians()) * i).toInt()
+                        ).toVec3d()
```
Comment (superblaubeere27) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt]: DRY

**Review by superblaubeere27 — COMMENTED** [review]
The vulcan tower implementation looks good.

I am not too sure about the expand implementation. It it not that well integrated with the other modes and the configuration is confusing. On the other hand it's pretty hard to patch something like expand into the current scaffold engine since everything it does is against the philosophy of it. I will have to think a bit to get ideas for improvement...

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt (hunk 1 above)
Comment (localpthebest) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt]: credit goes to izuna for that

Comment (levzzz5154) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt]: i think it should be expanding in the dirrection of player movement
also your impl doesn't support fractional expand lengths

Comment (levzzz5154) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt]: how could you misspell length 😭

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt (hunk 2 above)
Comment (levzzz5154) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt]: ye it certainly should be extracted into its own func

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt [hunk 3]
```diff
@@ -607,4 +632,65 @@ object ModuleScaffold : Module("Scaffold", Category.WORLD) {
...
+                            Hand.MAIN_HAND,
+                            BlockHitResult(
+                                block.interactedBlockPos.toVec3d(),
+                                block.direction,
+                                block.interactedBlockPos,
+                                true
+                            ),
+                            sequence
+                        )
+                    }
```
Comment (levzzz5154) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt:678]: ahem i think you might be placing multiple blocks per tick here which certainly is not a good idea

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt (hunk 1 above)
Comment (localpthebest) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt]: man fuck off

Comment (localpthebest) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/scaffold/ModuleScaffold.kt]: now that someone actually decides to make a much requested feature you shit on their spelling smh

**Comment by 1zun4** [comment]
I will not accept the Expand scaffold. Please revert these changes so I can at least merge the vulcan tower.

**Comment by localpthebest** [comment]
> I will not accept the Expand scaffold. Please revert these changes so I can at least merge the vulcan tower.

https://github.com/CCBlueX/LiquidBounce/pull/3622

**Comment by 1zun4** [comment]
replaced by https://github.com/CCBlueX/LiquidBounce/pull/3639/files

**Pull request closed without merging**